File: cli/cmd/system_vars.go
package cmd

import (
	"github.com/spf13/cobra"

	"openpa.local/cli/internal/output"
)

type systemVar struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	Value       string `json:"value"`
}

func newSystemVarsCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "system-vars",
		Short: "List the env vars OpenPA injects into exec_shell subprocesses",
		Long: `List the system variables registered on the OpenPA server.

These are the env vars (e.g. OPENPA_SERVER, OPENPA_TOKEN, OPENPA_SKILL_DIR)
that the server injects into every exec_shell subprocess. The values are
resolved server-side for the caller's profile (from the JWT), so the
output reflects exactly what an exec_shell run would see.`,
		RunE: func(cmd *cobra.Command, args []string) error {
			if err := requireToken(); err != nil {
				return err
			}
			ctx := cmd.Context()
			var vars []systemVar
			if err := state.Client.GetJSON(ctx, "/api/system-vars", &vars); err != nil {
				return err
			}
			if state.Output.JSON {
				return output.PrintJSON(vars)
			}
			t := output.NewTable(state.Output, "NAME", "VALUE", "DESCRIPTION")
			for _, v := range vars {
				t.AddRow(v.Name, v.Value, v.Description)
			}
			t.Render()
			return nil
		},
	}
}

File: app/cli/commands/system_vars.py
"""`opa system-vars` — list env vars OpenPA injects into exec_shell subprocesses.

Mirrors `cli/cmd/system_vars.go`.
"""

from __future__ import annotations

import typer

from app.cli.commands._helpers import graceful_errors


@graceful_errors
def system_vars(ctx: typer.Context) -> None:
    """List the env vars OpenPA injects into exec_shell subprocesses."""
    import asyncio

    from app.cli.client._base import Client
    from app.cli.client.system_vars import get_system_vars
    from app.cli.config import Config
    from app.cli.output import OutputMode, Table, print_json

    cfg: Config = ctx.obj["cfg"]
    mode: OutputMode = ctx.obj["mode"]
    cfg.require_token()

    async def _run() -> list:
        async with Client(cfg) as client:
            return await get_system_vars(client)

    vars_list = asyncio.run(_run())

    if mode.json:
        print_json([
            {"name": v.name, "description": v.description, "value": v.value}
            for v in vars_list
        ])
        return

    table = Table(mode, "NAME", "VALUE", "DESCRIPTION")
    for v in vars_list:
        table.add_row(v.name, v.value, v.description)
    table.render()
